pkg/server/server.go

```diff
@@ -0,0 +1,176 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+)
+
+var TILEROOT string = "/tiles/"
+var regions []string = []string{"iberia", "mediaeval_middle_east", "northern_europe"}
+var BELOWLOW string = "BL"
+var ABOVEHIGH string = "UH"
+var NOTILE string = "NT"
+
+func TileServer() {
+	c := cache.New(48*60*time.Minute, 60*time.Minute)
+	for _, region := range regions {
+		createRegion(c, region)
+	}
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", handle(c))
+	log.Fatal(http.ListenAndServe(":8000", mux))
+}
+
+func createRegion(c *cache.Cache, region string) {
+	dirs, err := ioutil.ReadDir(TILEROOT + region)
+	fmt.Printf("Loading: %s", region)
+	if err != nil {
+		return
+	}
+
+	var years []int
+
+	for _, year := range dirs {
+		yearNum, err := strconv.Atoi(year.Name())
+		if err == nil {
+			/**Avoid things that end in BC **/
+			years = append(years, yearNum)
+		}
+	}
+
+	sort.Ints(years)
+	c.Set(region, years, cache.NoExpiration)
+	return
+}
+
+func modifiedBinarySearch(years *[]int, year int, low int, high int) (int, int, error) {
+	/**get range between year that map to same tile**/
```

Comment:
Is there a reason for the double pointer at the top of the function? I would do `yearRange = *years` and use `yearRange` to keep things simple.